File: packages/stera_recorder/android/src/main/kotlin/open/fpvlabs/stera/ar_recorder/ArRecordingState.kt
package open.fpvlabs.stera.ar_recorder

/**
 * Enumeration of possible states for the AR recording session.
 */
enum class ArRecordingState {
    UNINITIALIZED,
    INITIALIZING,
    READY,
    RECORDING,
    PAUSED,
    STOPPING,
    CANCELLING,
    ERROR,
    DISPOSED;

    companion object {
        /**
         * Converts the enum to its string representation for Flutter serialization.
         */
        fun toString(state: ArRecordingState): String = state.name
    }
}

/**
 * Enumeration of ARCore tracking states.
 */
enum class ArTrackingState {
    TRACKING,
    PAUSED,
    STOPPED;

    companion object {
        fun fromArcoreState(state: com.google.ar.core.TrackingState?): ArTrackingState {
            return when (state) {
                com.google.ar.core.TrackingState.TRACKING -> TRACKING
                com.google.ar.core.TrackingState.PAUSED -> PAUSED
                com.google.ar.core.TrackingState.STOPPED -> STOPPED
                else -> STOPPED
            }
        }
    }
}

File: packages/stera_recorder/ios/stera_recorder/Sources/stera_recorder/ArRecordingState.swift
import Foundation

/// Enumeration of possible states for the AR recording session.
enum ArRecordingState: String {
    case uninitialized = "UNINITIALIZED"
    case initializing = "INITIALIZING"
    case ready = "READY"
    case recording = "RECORDING"
    case paused = "PAUSED"
    case stopping = "STOPPING"
    case cancelling = "CANCELLING"
    case error = "ERROR"
    case disposed = "DISPOSED"
}

/// Enumeration of ARKit tracking states.
enum ArTrackingState: String {
    case tracking = "TRACKING"
    case paused = "PAUSED"
    case stopped = "STOPPED"
}
